firmware/applications/SignBuddy/imu.c

```diff
@@ -0,0 +1,70 @@
+#include "imu.h"
+
+#include "board.h"
+#include "common.h"
+#include "i2c.h"
+#include "logger.h"
+
+#define PROCESS_PERIOD_MS    1000
+
+typedef struct {
+  uint32_t last_ticks;
+  i2c_t    i2c_instance;
+} state_t;
+
+static state_t s;
+
+static void hw_init(void)
```

Comment:
Should this be in i2c.c?

Reply:
no, I wanted it to be a generic config that you pass to i2c

Reply:
I was referring to the hw_init function, not sure if you were also referring to that

Reply:
i was, the hw_init is just a generic function to initialize things related to hardware for the imu module. I want the i2c module to be module agnostic and not a single instance module, therefore the imu module should pass in a config that gets applied by the i2c module

Reply:
I see now thanks for the clarification